public/app/features/live/pages/RuleModal.tsx

```diff
@@ -36,64 +145,27 @@ export const RuleModal: React.FC<Props> = (props) => {
         })}
       </TabsBar>
       <TabContent>
-        {activeTab === 'converter' && <ConverterEditor {...props} />}
-        {activeTab === 'processor' && <ProcessorEditor {...props} />}
-        {activeTab === 'output' && <OutputEditor {...props} />}
+        {success && <Alert title="Saved successfully" severity="success" onRemove={onRemoveSuccessAlert} />}
+        {error && <Alert title="Failed to save" severity="error" onRemove={onRemoveErrorAlert} />}
+        <SettingsEditor
+          {...props}
+          onBlur={onBlur}
```

Comment:
The behavior of this component is pretty confusing because you pass in different arguments than you get out.  Could this pass our an `onChange: (rule)`?  That way the internal detail that things are JSON strings is encapsulated in the SettingsEditor rather than both places.